ci/script.sh

```diff
@@ -7,13 +7,28 @@ set -ex
 # to target/$TARGET/{debug,release} which can reduce the number of needed conditionals in the
 # `before_deploy`/packaging phase
 
-cargo build --target $TARGET --verbose
+case $TARGET in
+  # use an emulator to run the cross compiled binaries
+  arm-unknown-linux-gnueabihf)
+    # build tests but don't run them
+    cargo test --target $TARGET --no-run
 
-# can't run binaries for this target
-if [ "$TARGET" != "arm-unknown-linux-gnueabihf" ]; then
-  cargo run --target $TARGET
-  cargo test --target $TARGET
-fi
+    # run tests in emulator
+    find target/$TARGET/debug -maxdepth 1 -executable -type f | \
+      xargs qemu-arm -L /usr/arm-linux-gnueabihf
```

Comment:
I have not tested it to be sure, but this seems wrong. `xargs` will pass all the files it finds as arguments to single invocation of `qemu-arm`, but `qemu-arm` expects an executable and arguments to pass to it, no? So only the first test will run this way.

To run each test separately, `xargs` is not needed. Instead just use find's `-exec`. Like:

    find target/$TARGET/debug -maxdepth 1 -executable -type f -exec qemu-arm -L /usr/arm-linux-gnueabihf '{}' ';'